grails-app/services/org/pih/warehouse/picklist/PicklistService.groovy

```diff
@@ -86,6 +87,34 @@ class PicklistService {
         productAvailabilityService.refreshProductsAvailability(orderItem?.order?.origin?.id, [orderItem?.product?.id], binLocations?.unique(), false)
     }
 
+    void clearPicklist(String id) {
+        Picklist picklist = Picklist.get(id)
+
+        if (!picklist) {
+            throw new ObjectNotFoundException(id, Picklist.class.toString())
+        }
+        // Store bin locations' id for product availability refresh
+        List<String> binLocations = []
+
+        Set<PicklistItem> picklistItems = picklist.picklistItems
+        List<PicklistItem> itemsToRemove = []
+        picklistItems.each { PicklistItem picklistItem ->
+            picklistItem.disableRefresh = true
+            itemsToRemove.add(picklistItem)
+            binLocations.add(picklistItem.binLocation?.id)
+        }
+        itemsToRemove.each {
+            picklist.removeFromPicklistItems(it)
+        }
+
+
+        productAvailabilityService.refreshProductsAvailability(
+                picklist?.requisition?.origin?.id,
+                picklist?.requisition?.requisitionItems?.product?.id,
```

Comment:
This is a bit misleading at first glance. I think it's doing this implicitly
```
picklist?.requisition?.requisitionItems?.collect { it?.product?.id } 
```
so let's make that explicit so it's slightly more readable and less confusing.
```
Set<String> productIds = picklist?.requisition?.requisitionItems?.collect { it?.product?.id } 
productAvailabilityService.refreshProductsAvailability(picklist?.requisition?.origin?.id,
                productIds, binLocations, false)
```
Note: The collect method might not allow the return to be a set, so if necessary turn it into a List and add .unique() at the end.

Reply:
Yeah, this is doing the `.collect` behind the scenes. I've just used "groovy magic" :D but I'm fine with doing it implicitly